File: fixtures/simple-fns/tests/bindings/test_simple_fns.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import uniffi_simple_fns

assert(getString() == "String created by Rust")
assert(getInt() == 1289)
assert(stringIdentity(s: "String created by Kotlin") == "String created by Kotlin")
assert(byteToU32(byte: 255) == 255)

let aSet = newSet()
addToSet(set: aSet, value: "foo")
addToSet(set: aSet, value: "bar")
assert(setContains(set: aSet, value: "foo"))
assert(setContains(set: aSet, value: "bar"))
assert(!setContains(set: aSet, value: "baz"))

File: fixtures/simple-fns/tests/bindings/test_simple_fns.rb
# frozen_string_literal: true

# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

require 'test/unit'
require 'uniffi_simple_fns'

class TestSimpleFns < Test::Unit::TestCase
  def test_simple_fns
    assert_equal 'String created by Rust', UniffiSimpleFns.get_string
    assert_equal 1289, UniffiSimpleFns.get_int
    assert_equal 'String created by Ruby', UniffiSimpleFns.string_identity('String created by Ruby')
    assert_equal 255, UniffiSimpleFns.byte_to_u32(255)

    a_set = UniffiSimpleFns.new_set
    UniffiSimpleFns.add_to_set a_set, 'foo'
    UniffiSimpleFns.add_to_set a_set, 'bar'

    assert UniffiSimpleFns.set_contains(a_set, 'foo')
    assert UniffiSimpleFns.set_contains(a_set, 'bar')
    assert !UniffiSimpleFns.set_contains(a_set, 'baz')

    assert_equal({'a' => 'b'}, UniffiSimpleFns.hash_map_identity({'a' => 'b'}))
  end
end
